doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,254 @@
+  * `<vch name>-SG-network-b`    dynamic membership: NIC-5 (of containerVM5), NIC-6 (of containerVM6)
+
+We apply the following security policies to our security groups:
+  * `<vch name>-SP-0` is applied to all the security groups
+  * `<vch name>-SP-1` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b` 
+  * `<vch name>-SP-2` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b`
+  * `<vch name>-SP-3` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b`
+
+The result we get :
+  * all the networks are isolated to each other
+  * containerVMs in the network can connect to their gateway at VCH
+
+## NSX Manager API:
+![Code flow to use NSX Manager API](pics/with-nsx.png)
+
+### API endpoint
+https://{nsxmanager's hostname or IP}/api 
+
+### API calls
+(details in http://pubs.vmware.com/nsx-63/topic/com.vmware.ICbase/PDF/nsx_63_api.pdf , note that many NSX API methods reference vCenter object IDs in URI parameters, query parameters, request bodies, and response bodies, also it takes XML as request body (sad) )
+* VCH create
+  * create security group 
+    * POST /2.0/services/securitygroup/bulk/{scopeId}
+  * create security policies and apply them to the security group
+    * POST /2.0/services/policy/securitypolicy (specify securityGroupBinding in the xml request body)
+
+* docker network create
+  * create security group `<vch name>-SG-<network name>`
+     * POST /2.0/services/securitygroup/bulk/{scopeId} (For the scopeId use globalroot-0 for non-universal security groups and universalroot-0 for universal security groups.)
+  * update security policies to apply them to the security group
+     * PUT /2.0/services/policy/securitypolicy/{ID}
+
+* docker network delete 
+  * check if any VMs attached to this security group
+     * GET /2.0/services/securitygroup/{objectId}/translation/virtualmachine 
+  * delete the network (security group, policy group)
+     * DELETE /2.0/services/securitytags/tag/{tagId}
+     * DELETE /2.0/services/securitygroup/{objectId} 
+
+* docker network list
+  * list all the security groups
+    * GET /2.0/services/securitygroup/internal/scope/{scopeId}
+  * filter security group names by `<vch name>-SG-` prefix
+
+* docker run --net=<network name> <image name>
+  * update security group `<vch name>-SG-<network name>` dynamic membership 
+    * PUT /2.0/services/securitygroup/bulk/{objectId}
+
+
+## Discussion:
+
+### Bridged Containers with Exposed Port or Directly Connected to Public network
+  * all the user cases mentioned in the networking design document should still be valid 
+
+## Unclear
+### Adding a ContainerVM to Multiple Networks
+This is not supported in current VIC implementation. 
+  * Is adding another NIC to a containerVM the right way to do it?
+  * Does having multiple interfaces matter ? How would user use multiple NICs?
```

Comment:
I feel like this overlaps with the "scopes" work @hmahmood did.  @hmahmood do we add a new scope for a new set of containers that doesn't exactly fit the smaller set or does the scope get extended?

Basically my question is whether or not the semantic is different than we currently have with scopes.